addons/website/static/src/js/tours/tour_manager.js

```diff
@@ -0,0 +1,26 @@
+/** @odoo-module **/
+
+import TourManager from 'web_tour.TourManager';
+
+TourManager.include({
+    /**
+     * @override
+     */
+    async _waitBeforeTourStart() {
+        const res = await this._super(...arguments);
+        if (window.location.href.indexOf('/web#action=website.website_preview') > 0) {
+            // The tour manager waits for the final WebsitePreview client
+            // action's url to start the tour, in order to avoid reloading the
+            // browser if the it is already at the right location.
+            await new Promise(resolve => {
+                this.clientActionUrlInterval = setInterval(() => {
+                    if (window.location.pathname.startsWith('/@/')) {
+                        resolve();
+                        clearInterval(this.clientActionUrlInterval);
+                    }
+                }, 100);
+            });
+        }
```

Comment:
I know it is what I suggested as a worst case possibility but no other way? Cannot listen for the URL change to /@/ for example?

Reply:
(did you check if you actually enters this `if`? If yes, it would be sad to lose 100ms x number of tours)

Reply:
Yes, it enters the loop for every website* test (and wait for 100ms to get the '/@/' pathname). I guess the file could define an override for the `WebsitePreview._replaceBrowserUrl` that would send an event when the frontend url is set ?

Reply:
> Cannot listen for the URL change to /@/ for example?

:point_up: ? But yes, a specific event trigger could do the trick too yes (but could trigger it in the method's original definition in website rather than doing an override here IMO)

Reply:
Unfortunately the `popstate` and `hashchange` events are not triggered when using `pushState` or `replaceState`, so we have to come up with our own event for that I think

Reply:
Out of curiosity, can I ask why it would be better to define that event in the original method ?
I am asking because the event has to be sent to the legacy core bus in order for the legacy TourManager to receive it, and I would think it is better to avoid including legacy-related logic in the original WebsitePreview definition

Reply:
Do as you like :wink: It's just my opinion that an app making a patch to its own classes in different file is not good. Less efficient, less readable, more error prone in refactoring, I cannot think of an advantage :shrug:

Reply:
Changed to send the event from the original WebsitePreview class :+1:
This should be ready for review